.github/workflows/devcontainer.yaml

```diff
@@ -0,0 +1,47 @@
+name: devcontainer
+
+on:
+  push:
+    paths:
+      - .devcontainer/base/Dockerfile
+      - .devcontainer/base/devcontainer.json
+  pull_request:
+    paths:
+      - .devcontainer/base/Dockerfile
+      - .devcontainer/base/devcontainer.json
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - id: set-platforms
+        run: |
+          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
+            echo "platforms=linux/amd64,linux/arm64" >>$GITHUB_OUTPUT
+          else
+            echo "platforms=linux/amd64" >>$GITHUB_OUTPUT
+          fi
+
+      - name: Pre-build dev container image
+        uses: devcontainers/ci@v0.2
+        with:
+          subFolder: .devcontainer/base
```

Comment:
It does not seem to be possible to configure this way; I have created an issue.
devcontainers/ci#212

Reply:
Hmmm. I wonder whether this is substantially better than using a normal docker workflow like we have now. Because I _think_ the only thing that's pushed is the output of the Docker build. Is that right?

If it is, it might be that we can mostly swap out the existing Dockerfile for the new one, and use the existing workflow...

Reply:
> I wonder whether this is substantially better than using a normal docker workflow like we have now. Because I _think_ the only thing that's pushed is the output of the Docker build. Is that right?

Since Dev Container Features are a fairly long shell script, I think it provides a better setup than simply installing Python, etc.
Dev Container CLI is required to build a Docker image with Dev Container Features installed.

But as you point out, it doesn't support all the features of `buildx build` (such as GitHub Actions cache support), so it's worth moving to a simple Docker file-only build.
Is that better for you?

Reply:
Oh I think I see now — this will cache running the steps in `.devcontainer/devcontainer.json` as well as the Dockerfile? That's excellent if so!

In that case I would probably keep it as-is, because then folks don't need to do any building themselves, they get a fully baked image...